src/librustc_builtin_macros/asm.rs

```diff
@@ -16,7 +16,7 @@ struct AsmArgs {
     named_args: FxHashMap<Symbol, usize>,
     reg_args: FxHashSet<usize>,
     options: ast::InlineAsmOptions,
-    options_span: Option<Span>,
+    options_spans: Option<Vec<Span>>,
```

Comment:
Should I do an `assert!(options_spans.len() > 0)` where there's currently an unwrap? Maybe it's not necessary, but it might make it easier to debug in the future if the code there changes

Reply:
I don't know what the compiler etiquette is in these situations, so I wanted to check

Reply:
Well an empty `Vec` would simply mean that no options were specified, in which case none of the error paths that use `options_spans` should be hit. I wouldn't bother with an assert in this case.

Reply:
> A good place to add tests would be `src/test/codegen/asm-options.rs`.

I've never written a codegen test before, so could you suggest how to write it or point me toward a description of how to write one? Thanks!

Reply:
Just copy the existing ones but split the options across multiple `options()` blocks.

Reply:
Okay, how's this? 49190a2